pkg/utils/utils.go

```diff
@@ -445,3 +456,189 @@ func DoesMigrationScheduleMigrateNamespaces(migrationSchedule stork_api.Migratio
+				return uid, gid, nil
+			}
+		}
+	} else {
+		// Only one container for the pod is found, let's pick the UID or GID of it
+		securityContext := pod.Spec.Containers[0].SecurityContext
+		uid, gid := getIdFromSecurityContext(securityContext)
+		if uid != UndefinedId || gid != UndefinedId {
+			return uid, gid, nil
+		}
+	}
+	// UID or GID is not found in any of the container spec, try Getting the UID/GID from the pod's security context
+	securityContext := pod.Spec.SecurityContext
+	uid, gid = getIdFromSecurityContext(securityContext)
+	if uid != UndefinedId || gid != UndefinedId {
+		return uid, gid, nil
+	}
+	// Neither pod nor container has the securityContext defined with User Id or group Id, return error with undefined uids
+	return uid, gid, fmt.Errorf("failed to get user id or group id from securityContext of pod [%s/%s]", pod.Namespace, pod.Name)
+}
+
+// Get uid gid from application backup cr
+func GetUIDGIDFromBackupCR(backup *stork_api.ApplicationBackup, pvcName string) (int64, int64, error) {
+	fn := "GetUIDGIDFromBackupCR"
+	// read the uid and gid from the application backup cr's volume array
+	if backup != nil {
+		for _, volume := range backup.Status.Volumes {
+			if volume.PersistentVolumeClaim == pvcName {
+				logrus.Infof("%v: Found uid:[%v] and gid:[%v] for the restore of the pvc[%v]", fn, volume.VolumeJobSecurityContext.RunAsUser, volume.VolumeJobSecurityContext.RunAsGroup, volume.PersistentVolumeClaim)
+				return volume.VolumeJobSecurityContext.RunAsUser, volume.VolumeJobSecurityContext.RunAsGroup, nil
+			}
+		}
+	}
+	err := fmt.Errorf("failed to obtain uid from backup cr")
+	return UndefinedId, UndefinedId, err
+}
+
+// GetPsaEnabledAppUID - Get the UID/GID of the application pod which has psa enforcement in its namespace using the PVC
+func GetPsaEnabledAppUID(pvcName string, namespace string, backup *stork_api.ApplicationBackup, getUIDFromApp bool) (int64, int64, bool, error) {
+	fn := "GetPsaEnabledAppUID"
+	uid := UndefinedId
+	gid := UndefinedId
+	psaIsEnforced := false
+
+	// Check if the namespace has the PSA labels set
+	psaIsEnforced, psaLabelMap, err := GetPsaDetail(namespace)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return uid, gid, psaIsEnforced, err
+	}
+	if !psaIsEnforced {
+		errMsg := fmt.Sprintf("PSA is not enforced in namespace [%s], no need to enforce any uid/gid in job for backing up pvc [%s]", namespace, pvcName)
+		logrus.Debugf("%s: %v", fn, errMsg)
+		return uid, gid, psaIsEnforced, fmt.Errorf(errMsg)
+	}
+
+	// check in the psa label map if the pod has the psa enforcement labels set to restricted
+	for k, v := range psaLabelMap {
+		// Only in case of restricted enforcement, we need to get the UID/GID of the pod
+		// otherwise baseline and privilege will work if the job pod runs with root access.
+		// TODO: Should we preserve the UID in case baseline config too, though not required for JOB.
+		if strings.Contains(k, "enforce") && v == "restricted" {
```

Comment:
Small ehancement was we change the following where `psaLabelMap[k] = v` assignment happens only in `podSecurityStandardEnforceMode` that way map will have only one entry always.
Just avoiding one more chedk here.

```
for k, v := range ns.GetLabels() {
		// Use the regular expression to match the label key
		if pattern.MatchString(k) {
			
			if k == podSecurityStandardEnforceMode {
                                 psaLabelMap[k] = v
				IsPsaEnforced = true
			}
		}
	}
```

Reply:
Prasanth while I appreciate your concerned point here but I would still want to keep the map to be populated with all the values. and it is at max can grow to 6 entries as of this version for PSA. And this can help in future if for certain type of standard some default logic changes then we have to alter things here. So better to have a map remembering all setting. 

I assume you are suggesting to store to map only when it is Enforced...and that way we can save a if check down below. but I am pretty sure other data of the map will soon be needed as soon as we bring in OCP and other managed clusters. And on top of that it is not that of a heavy duty map from cpu/memory wise.